### ISSUE TITLE
Problems found in recategorization

### ISSUE THREAD

**Issue by ntBre** [astral-sh/ruff#28070]
This is a catch-all issue for problems we noticed while going through all the rules for recategorization.

## Bugs

General bugs

- [global-variable-not-assigned (PLW0602)](https://docs.astral.sh/ruff/rules/global-variable-not-assigned/#global-variable-not-assigned-plw0602) flags read-only cases where `global` is used to skip an enclosing scope:

	```py
	DEBUG = True
	
	def outer:
	   DEBUG = False
		def foo():
		    global DEBUG
		    if DEBUG:
		        print("foo() called")
	```

- [import-self (PLW0406)](https://docs.astral.sh/ruff/rules/import-self/#import-self-plw0406) has some false positives in the ecosystem:
  - https://github.com/Snowflake-Labs/snowcli/blob/68e15d26a854c38e9ae2cf2c038fd3810d8cc79a/tests/test_experimental_behaviour.py#L59
  - https://github.com/mlflow/mlflow/blob/966daa9ad1ddb66cade6ee867b11d8279fd33a51/mlflow/genai/scorers/builtin_scorers.py#L353
  - https://github.com/pypa/pip/blob/fc24f1eae6c3e2926874be2444232653d98d3075/src/pip/_vendor/rich/box.py#L428

  Possibly these are going to be too tricky for us to understand, but they do seem to be false positives

- [redefined-while-unused (F811)](https://docs.astral.sh/ruff/rules/redefined-while-unused/#redefined-while-unused-f811) has false positives in notebooks and in pytest fixtures (621 ecosystem hits, most(?) of which are wrong)

  Actually the ones I clicked on outside of notebooks don't appear to be wrong. Part of this point will be fully analyzing the ecosystem results since I didn't find any great examples to pull out immediately.

## Notebook bugs

Bugs specifically confined to notebooks

- [undefined-local (F823)](https://docs.astral.sh/ruff/rules/undefined-local/#undefined-local-f823)
  - https://github.com/zanieb/huggingface-notebooks/blob/68cd6fa1a2831c5189f85257c13d691cb76292db/transformers_doc/en/asr.ipynb
- [late-future-import (F404)](https://docs.astral.sh/ruff/rules/late-future-import/#late-future-import-f404)
  - https://github.com/openai/openai-cookbook/blob/4a85c3018d20ceef48bf7549450c567896501bf9/examples/agents_sdk/agent_improvement_loop.ipynb

    I think the problem here is the preceding cell with only ipython magic commands in it.

## Rules that should be split into multiple rules

These are cases where part of a rule is enforcing an issue with higher severity than other parts of the rule, e.g. the rule flags both `correctness` issues and `suspicious` or even `style` issues. It should be split into separate rules for each part.

- [is-literal (F632)](https://docs.astral.sh/ruff/rules/is-literal/#is-literal-f632) flags cases that are `SyntaxWarning`s (`correctness`) but also cases that are not (`suspicious`)
- [undefined-export (F822)](https://docs.astral.sh/ruff/rules/undefined-export/#undefined-export-f822) flags runtime `NameError`s (`correctness`) but in preview also flags names that may correspond to modules in the same package (`suspicious`)
- [loop-iterator-mutation (B909)](https://docs.astral.sh/ruff/rules/loop-iterator-mutation/#loop-iterator-mutation-b909) flags mutations of all collection types, only some of which (sets) are definite errors. This makes it currently a `suspicious` rule but some part of it could likely be `correctness`

## Rules that should be combined into one rule

In their current forms, these rules typically have slightly different coverage but conceptually catch the same issue.

- [bad-str-strip-call (PLE1310)](https://docs.astral.sh/ruff/rules/bad-str-strip-call/#bad-str-strip-call-ple1310) and [strip-with-multi-characters (B005)](https://docs.astral.sh/ruff/rules/strip-with-multi-characters/#strip-with-multi-characters-b005)
- [modified-iterating-set (PLE4703)](https://docs.astral.sh/ruff/rules/modified-iterating-set/#modified-iterating-set-ple4703) and [loop-iterator-mutation (B909)](https://docs.astral.sh/ruff/rules/loop-iterator-mutation/#loop-iterator-mutation-b909)
  This somewhat contradicts the split suggestion above, but in their current form, these two rules overlap heavily.

## Documentation

These rules are functionally fine but have some issues in their documentation.

- [logging-eager-conversion (RUF065)](https://docs.astral.sh/ruff/rules/logging-eager-conversion/#logging-eager-conversion-ruf065)

  The example shows a string as input, which isn't a very interesting conversion. It could also be better clarified if this rule is mostly about performance or the complexity of the code.

- [f-string (UP032)](https://docs.astral.sh/ruff/rules/f-string/#f-string-up032)

  We categorized this as `complexity` but the docs emphasize something closer to `style`.

- [single-item-membership-test (FURB171)](https://docs.astral.sh/ruff/rules/single-item-membership-test/#single-item-membership-test-furb171)

  The example here is pretty silly, using literals everywhere.
